src/Microsoft.TestPlatform.Common/ExtensionFramework/VSExtensionManager.cs

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Reflection;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.Utilities.Helpers.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.Utilities.Helpers;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+    using Microsoft.VisualStudio.TestPlatform.PlatformAbstractions;
+    using Microsoft.VisualStudio.TestPlatform.Common.Utilities;
+    using Microsoft.VisualStudio.TestPlatform.Common.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.Common.Resources;
+
+    public class VSExtensionManager : IVSExtensionManager
+    {
+        private const string PrivateAssembliesDirName = "PrivateAssemblies";
+        private const string PublicAssembliesDirName = "PublicAssemblies";
+        private const string ExtensionManagerService = "Microsoft.VisualStudio.ExtensionManager.ExtensionManagerService";
+        private const string ExtensionManagerAssemblyName = @"Microsoft.VisualStudio.ExtensionManager";
+        private const string ExtensionManagerImplAssemblyName = @"Microsoft.VisualStudio.ExtensionManager.Implementation";
+        private const string DevenvExe = "devenv.exe";
+
+        private const string SettingsManagerTypeName = "Microsoft.VisualStudio.Settings.ExternalSettingsManager";
+        private const string SettingsManagerAssemblyName = @"Microsoft.VisualStudio.Settings.15.0";
```

Comment:
How do we manage this version for dev16?

Reply:
Aren't we thinking of deprecating this in the next release? @sudiptadmsft

Reply:
Awesome.